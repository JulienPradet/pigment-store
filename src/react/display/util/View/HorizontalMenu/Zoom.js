import React from 'react'
import classnames from 'classnames'
import {withHandlers} from 'recompose'
import Slider from 'rc-slider'
import './slider.css'
import styles from './horizontalmenu.css'

const Zoom = (
  withHandlers({
    onChange: ({onChange}) => (value) => onChange({zoom: value})
  })
)(({zoom, onChange}) => <div className={styles.zoom}>
  <div className={styles.zoomItem}>
    Zoom
    <input type='number' step='5' value={zoom} onChange={(event) => onChange(event.currentTarget.value)} />
    %
  </div>
  <div className={classnames(styles.zoomItem, styles.zoomSlider)}>
    <Slider
      min={25}
      max={400}
      value={parseInt(zoom)}
      onChange={onChange}
      tipFormatter={null}
    />
  </div>
</div>)

Zoom.displayName = 'Zoom'

export default Zoom
