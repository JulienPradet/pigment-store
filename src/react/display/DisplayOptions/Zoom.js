import React from 'react'
import {Group, Item} from '../util/View/HorizontalMenu'

const Zoom = ({zoom, onChange}) => <Group>
  <Item>
    <input type='number' step='5' value={zoom} onChange={(event) => onChange({ zoom: event.currentTarget.value })} />
    %
  </Item>
</Group>

export default Zoom
