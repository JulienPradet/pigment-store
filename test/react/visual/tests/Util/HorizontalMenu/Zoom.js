import React from 'react'
import PigmentStore from 'pigment-store'
import Zoom from '../../../../../../src/core/display/App/util/View/HorizontalMenu/Zoom'

const aZoomConfig = {
  zoom: 100,
  onChange: function onChange (newZoom) {
    console.log(newZoom)
  }
}

export default PigmentStore.React.describe('Zoom', Zoom)
  .featureJsx(
    'Default',
    <Zoom {...aZoomConfig} />
  )
  .featureJsx(
    'With max value',
    <Zoom {...aZoomConfig} zoom={200} />
  )
  .featureJsx(
    'With min value',
    <Zoom {...aZoomConfig} zoom={10} />
  )
