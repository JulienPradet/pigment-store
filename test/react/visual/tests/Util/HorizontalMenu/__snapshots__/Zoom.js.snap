exports[`Zoom Default 1`] = `
<div
  className="zoom">
  <div
    className="zoomItem">
    Zoom
    <input
      onChange={[Function]}
      step="5"
      type="number"
      value={100} />
    %
  </div>
  <div
    className="zoomItem zoomSlider">
    <div
      className="rc-slider"
      onMouseDown={[Function]}
      onTouchStart={[Function]}>
      <div
        className="rc-slider-rail" />
      <div
        className="rc-slider-track rc-slider-track-1"
        style={
          Object {
            "left": "0%",
            "visibility": "visible",
            "width": "20%",
          }
        } />
      <div
        className="rc-slider-step" />
      <div
        className="rc-slider-handle rc-slider-handle-2 rc-slider-handle-upper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "left": "20%",
          }
        } />
      <div
        className="rc-slider-mark" />
    </div>
  </div>
</div>
`;

exports[`Zoom With max value 1`] = `
<div
  className="zoom">
  <div
    className="zoomItem">
    Zoom
    <input
      onChange={[Function]}
      step="5"
      type="number"
      value={200} />
    %
  </div>
  <div
    className="zoomItem zoomSlider">
    <div
      className="rc-slider"
      onMouseDown={[Function]}
      onTouchStart={[Function]}>
      <div
        className="rc-slider-rail" />
      <div
        className="rc-slider-track rc-slider-track-1"
        style={
          Object {
            "left": "0%",
            "visibility": "visible",
            "width": "46.666666666666664%",
          }
        } />
      <div
        className="rc-slider-step" />
      <div
        className="rc-slider-handle rc-slider-handle-2 rc-slider-handle-upper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "left": "46.666666666666664%",
          }
        } />
      <div
        className="rc-slider-mark" />
    </div>
  </div>
</div>
`;

exports[`Zoom With min value 1`] = `
<div
  className="zoom">
  <div
    className="zoomItem">
    Zoom
    <input
      onChange={[Function]}
      step="5"
      type="number"
      value={10} />
    %
  </div>
  <div
    className="zoomItem zoomSlider">
    <div
      className="rc-slider"
      onMouseDown={[Function]}
      onTouchStart={[Function]}>
      <div
        className="rc-slider-rail" />
      <div
        className="rc-slider-track rc-slider-track-1"
        style={
          Object {
            "left": "0%",
            "visibility": "visible",
            "width": "0%",
          }
        } />
      <div
        className="rc-slider-step" />
      <div
        className="rc-slider-handle rc-slider-handle-2 rc-slider-handle-upper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "left": "0%",
          }
        } />
      <div
        className="rc-slider-mark" />
    </div>
  </div>
</div>
`;
